UnityProject/Assets/Scripts/Chemistry/Reaction.cs

```diff
@@ -33,28 +34,62 @@ public bool Apply(MonoBehaviour sender, ReagentMix reagentMix)
 			{
 				return false;
 			}
-			var reactionAmount = ingredients.Min(i => reagentMix[i.Key] / i.Value);
-
-			if (!catalysts.All(catalyst =>
-				reagentMix[catalyst.Key] > catalyst.Value * reactionAmount))
+			if (useExactAmounts == false)
 			{
-				return false;
-			}
+				var reactionAmount = ingredients.Min(i => reagentMix[i.Key] / i.Value);
 
-			foreach (var ingredient in ingredients)
-			{
-				reagentMix.Subtract(ingredient.Key, reactionAmount * ingredient.Value);
-			}
+				if (!catalysts.All(catalyst =>
+					reagentMix[catalyst.Key] > catalyst.Value * reactionAmount))
+				{
+					return false;
+				}
 
-			foreach (var result in results)
-			{
-				var reactionResult = reactionAmount * result.Value;
-				reagentMix.Add(result.Key, reactionResult);
-			}
+				foreach (var ingredient in ingredients)
+				{
+					reagentMix.Subtract(ingredient.Key, reactionAmount * ingredient.Value);
+				}
 
-			foreach (var effect in effects)
+				foreach (var result in results)
+				{
+					var reactionResult = reactionAmount * result.Value;
+					reagentMix.Add(result.Key, reactionResult);
+				}
+
+				foreach (var effect in effects)
+				{
+					effect.Apply(sender, reactionAmount);
+				}
+			}
+			else if (useExactAmounts == true)
```

Comment:
90% of this else is just copypaste of if statement
Can't you just insert
`if (useExactAmounts && !ingredients.All(ingredient => reagentMix[ingredient.Key] >= ingredient.Value))
    return;`
after if statement at line 41 not to make unneeded nesting?